messaging-modules/rabbitmq/src/test/java/com/baeldung/benchmark/queue/dynamic/DynamicQueueCreationLiveTest.java

```diff
@@ -0,0 +1,71 @@
+package com.baeldung.benchmark.queue.dynamic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+public class DynamicQueueCreationLiveTest {
+
+    private static final String QUEUE_NAME = "baeldung-queue";
+    private static final String QUEUE_NAME_NEW = "baeldung-queue-new";
+
+    private static Connection connection;
+
+    @BeforeAll
+    public static void setUpConnection() throws IOException, TimeoutException {
+        ConnectionFactory factory = new ConnectionFactory();
```

Comment:
Does the rabbit queue have to be started?
If yes, it would be better to use some TestContainers.